src/modules/iasql_functions/rpcs/iasql_get_sql_since.ts

```diff
@@ -0,0 +1,342 @@
+  if (v && typeof v === 'object' && Array.isArray(v)) return `'{${v.map(o => getDbString(o)).join(',')}}'`;
+  return `${v}`;
+}
+
+/**
+ * @internal
+ * Returns Typeorm metadata related to `tableName`
+ */
+async function getMetadata(
+  tableName: string,
+  modsIndexedByTable: { [key: string]: ModuleInterface },
+  orm: TypeormWrapper,
+): Promise<EntityMetadata | RelationMetadata | undefined> {
+  const mappers = Object.values(modsIndexedByTable[tableName] ?? {}).filter(val => val instanceof MapperBase);
+  let metadata: EntityMetadata | RelationMetadata | undefined;
+  for (const m of mappers) {
+    const tableEntity = (m as MapperBase<any>).entity;
+    const entityMetadata = await orm.getEntityMetadata(tableEntity);
+    if (entityMetadata.tableName === tableName) {
+      metadata = entityMetadata;
+      break;
+    } else {
+      if (
+        entityMetadata.ownRelations
+          .filter(or => !!or.joinTableName)
+          .map(or => or.joinTableName)
+          .includes(tableName)
+      ) {
+        metadata = entityMetadata.ownRelations.find(or => or.joinTableName === tableName);
+        break;
+      }
+    }
+  }
+  // If no metadata found, we need to do a second pass over the mappers because it could be the case of an
+  // Entity that does not have it's own mapper but it is managed by another Entity Mapper.
+  if (!metadata) {
+    for (const m of mappers) {
+      const tableEntity = (m as MapperBase<any>).entity;
+      const entityMetadata = await orm.getEntityMetadata(tableEntity);
+      if (
+        entityMetadata.ownRelations
+          .filter(or => !!or.inverseEntityMetadata.tableName)
+          .map(or => or.inverseEntityMetadata.tableName)
+          .includes(tableName)
+      ) {
+        metadata = entityMetadata.ownRelations.find(
+          or => or.inverseEntityMetadata.tableName === tableName,
+        )?.inverseEntityMetadata;
+        break;
+      }
+    }
+  }
+  return metadata;
+}
+
+/**
+ * @internal
+ * Returns sub-query based on `id` relation.
+ * The related entity will be found using the cloud columns decorators.
+ */
+async function recreateSubQuery(
+  id: number,
+  entityMetadata: EntityMetadata,
+  modsIndexedByTable: { [key: string]: ModuleInterface },
+  orm: TypeormWrapper,
+): Promise<string> {
+  // Get cloud columns of the entity we want to look for.
+  const cloudColumns = getCloudId(entityMetadata?.target);
+  if (cloudColumns && !(cloudColumns instanceof Error)) {
+    let e: any;
+    try {
+      e = await orm.findOne(entityMetadata?.targetName ?? '', { where: { id } });
+    } catch (e: any) {
+      logger.warn(e.message ?? 'Error finding relation');
+      e = null;
+    }
+    const values = await Promise.all(
+      cloudColumns.map(
+        async (cc: string) =>
+          await jsonValueToDbString(entityMetadata.tableName, cc, e[cc], modsIndexedByTable, orm),
+      ),
+    );
+    return `(SELECT id FROM ${entityMetadata?.tableName} WHERE ${cloudColumns
```

Comment:
It's not *always* `id` as the join column name, though. Should we pull that from the entity, too?

Reply:
why do you think it is necessary for other join column names? As I thought about it, the database `id` is the only one that is worth updating with its subquery because it is autogenerated, it will not match across databases. Other columns like `load_balancer_name` (if we were joined by it) are up to the user to update them.